### PR TITLE
add missing dependencies for owl 0.3.7

### PR DESCRIPTION
**Pull request by mor1**

### CODE REVIEW

**Comment by mor1** [comment]
Hm. Travis failing on 4.06.0 because eigen fails because ctypes is being downgraded for some reason:   `↘  downgrade ctypes   0.14.0 to 0.4.2       [required by owl]`

File: packages/owl/owl.0.3.7/opam [hunk 1]
```diff
@@ -18,13 +18,16 @@ build-test: [
 
 depends: [
   "alcotest" {test}
+  "base"
   "base-bigarray"
   "conf-openblas" {>= "0.2.0"}
+  "configurator"
```
Comment (rgrinberg) [inline packages/owl/owl.0.3.7/opam]: This is likely a build dependency btw.

Comment (mor1) [reply packages/owl/owl.0.3.7/opam]: good point, thanks!

**Comment by kit-ty-kate** [comment]
> Hm. Travis failing on 4.06.0 because eigen fails because ctypes is being downgraded for some reason: ↘ downgrade ctypes 0.14.0 to 0.4.2 [required by owl]

Then I suppose a lower bound should be added to `ctypes`.

Also are all those dependencies needed for `configurator` or are they used for the library itself ? In the former case you might want to flag them as build dependencies as well.

File: packages/owl/owl.0.3.7/opam (hunk 1 above)
Comment (mseri) [reply packages/owl/owl.0.3.7/opam]: `configurator`, `base` and `stdio` seems to be used only in `src/owl/config/configure.ml`, I think they are all build dependencies

**Comment by pmetzger** [comment]
Just a note: the commit message here isn't very good. It should probably be "owl: add missing dependencies" or some such. Commits in git are to the whole repository, and thus it is difficult, just looking at the log, to know what they are for without more specificity.

**Comment by mor1** [comment]
@pmetzger Yes sorry, I'm normally careful to do that. Was careless last night/first thing this morning...

**Comment by mseri** [comment]
@mor1 `configurator`, `base` and `stdio` seems to be used only in `src/owl/config/configure.ml`, I think they are all just `build` dependencies

**Comment by kit-ty-kate** [comment]
There is a build failure in your tests on MacOS:
```
#=== ERROR while installing owl.0.3.7 =========================================#
# opam-version 1.2.2
# os           darwin
# command      jbuilder runtest -p owl -j 4
# path         /Users/travis/.opam/system/build/owl.0.3.7
# compiler     system (4.06.1)
# exit-code    1
# env-file     /Users/travis/.opam/system/build/owl.0.3.7/owl-22697-f8ee95.env
# stdout-file  /Users/travis/.opam/system/build/owl.0.3.7/owl-22697-f8ee95.out
# stderr-file  /Users/travis/.opam/system/build/owl.0.3.7/owl-22697-f8ee95.err
### stderr ###
# File "test/jbuild", line 7, characters 4-12:
# Error: Library "owl_base" not found.
# Hint: try: jbuilder external-lib-deps --missing -p owl @runtest
```

**Comment by mor1** [comment]
Thanks -- yes, I fixed that already (forgot to update `test/jbuild`) but haven't issued a release yet. Is it better for me to update the current pending release (0.3.7) artefacts or to issue a new one (0.3.8)?

**Comment by kit-ty-kate** [comment]
If nobody-else is using `0.3.7` yet then updating this PR would be good.

**Comment by mor1** [comment]
This looks to me like it's working as well as it can at the moment -- the failures seem to be Alpine 3.6 and Centos again (usual m4 and solver problems).

This might actually be ready to merge :)

**Comment by kit-ty-kate** [comment]
Seems like it doesn't compile on arm64 but it can be fix/constrained later. Thanks !

**Comment by mseri** [comment]
Travis is failing to run tests on ocaml 4.06 but I think we can look into it in a second stage
```
- File "test/jbuild", line 7, characters 4-7:
- Error: Library "owl" not found.
- Hint: try: jbuilder external-lib-deps --missing -p owl-base @runtest
[ERROR] The compilation of owl-base failed at "jbuilder runtest -p owl-base -j
        4".
#=== ERROR while installing owl-base.0.3.7 ====================================#
# opam-version 1.2.2
# os           darwin
# command      jbuilder runtest -p owl-base -j 4
# path         /Users/travis/.opam/system/build/owl-base.0.3.7
# compiler     system (4.06.1)
# exit-code    1
# env-file     /Users/travis/.opam/system/build/owl-base.0.3.7/owl-base-10920-5d7197.env
# stdout-file  /Users/travis/.opam/system/build/owl-base.0.3.7/owl-base-10920-5d7197.out
# stderr-file  /Users/travis/.opam/system/build/owl-base.0.3.7/owl-base-10920-5d7197.err
### stderr ###
# File "test/jbuild", line 7, characters 4-7:
# Error: Library "owl" not found.
# Hint: try: jbuilder external-lib-deps --missing -p owl-base @runtest

```

**Comment by kit-ty-kate** [comment]
Oh crap, I don't know how I missed that. Sorry.

**Comment by mor1** [comment]
I suspect the way tests are setup, they only apply to the owl package, not
owl-base. Do I need to make some changes to the opam files to stop tests
being run?

**Comment by kit-ty-kate** [comment]
yes please

**Comment by mseri** [comment]
You will need to remove the `build-test` Section from all the opam files that should not run tests

**Comment by mor1** [comment]
Done in #11869

**Pull request merged**